tests/test_edge_cases.py

```diff
@@ -1892,3 +1893,60 @@ class Config:
             arbitrary_types_allowed = True
 
     assert Model().x == Foo()
+
+
+def test_config_partial():
+    class Foo(BaseModel):
+        a: str = Field(...)
+        b: str = Field(...)
+        c: Optional[str] = Field()
+
+    Bar = create_partial_model(Foo)
+
+    # Ensure partial is only set on the new class
+    assert Foo.Config.partial is False
+    assert Bar.Config.partial is True
+
+    # Verify b (a required field) can be omitted on the partial model
+    b = Bar(a='x', c=None)
+    assert b.dict() == {'a': 'x', 'c': None}
```

Comment:
For reference as I mentioned it in [my PR](https://github.com/samuelcolvin/pydantic/pull/3121#issuecomment-925955894) : having `b` not appear even though we did not set `exclude_unset` is a surprising behavior to me, but it might be because I implemented it in a different way which shaped my expectations differently

Reply:
@christophelec yeah as I mentioned [here](https://github.com/samuelcolvin/pydantic/pull/3121#issuecomment-925960693), it can be easily changed. It just made more sense to me in the context of a "partial" model coming from a Django REST Framework background.

Reply:
@christophelec also just to flesh out the logic behind not showing the fields a bit more:

Given the model:
```python
class Foo(BaseModel):
    a: str
    b: Optional[str]
```

Looking at this, it's clear that `a` will always have at least a string value, and `b` may or may not.

As such, it's a bit misleading to ever assign `None` to a, if we were to include it:
```
f = Foo(b='hi')

# It feels strange / risky to assign None to a value that normally could not be None, based on the model's definition
f.dict()  # {'a': None, 'b': 'hi'}
```

Granted yes, the user should be using `exclude_unset=True` _anyway_, but returning None on a non-nullable field still feels like an impractical decision. Just my two cents though! Happy to change it if others disagree!